### PR TITLE
Renewal item db update

### PR DESCRIPTION
**Pull request by zackdreaver**
About this PR:
- Item script cleanup and fixes based of kRO server
- Update item combo scripts
- New items from kRO and jRO with custom Aegis names

### CODE REVIEW

**Comment by zackdreaver** [comment]
I'm kinda confuse with this script's documentation:
`bonus2 bSkillUseSPrate,sk,n;  Decreases SP consumption of skill sk by n%`

If I want to reduce sp cost of certain skill,
should I set it to positive or negative value,
as i see the similiar script `bonus2 bSkillUseSP,sk,n;` have negative value.

**Comment by sigtus** [comment]
Hello, zackdreaver.
First of all, I'm a bit confused about this one, for example:
`bonus bBaseAtk,(.@r*.@r)*(5/4);`
It would be the same as
`bonus bBaseAtk,.@r*.@r;`
Because the division 5/4 will return 1.
Is this what you're looking for?
`bonus bBaseAtk,powe(.@r,2)*5/4;`
Regards,

**Comment by zackdreaver** [comment]
(Refine level*Refine level)*1.25

**Comment by cydh** [comment]
r = r * r; r *= 125/100;

On Jan 4, 2017 15:23, "zackdreaver" <notifications@github.com> wrote:

(Refine level*Refine level)*1.25

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/rathena/rathena/pull/1848#issuecomment-270317885>, or mute
the thread

**Comment by zackdreaver** [comment]
I'm done with the commits.

**Comment by zackdreaver** [comment]
I'm done with the commits.
Special thanks to NovaRO for their patch notes.

**Comment by Akkarinage** [comment]
It'd be great if you could link your sources for this information.

**Comment by zackdreaver** [comment]
I'll write it twice then
- Item script cleanup and fixes based of kRO server

**Comment by Akkarinage** [comment]
I meant if there are sources on the web to link to them, and if they're direct from kRO whether you used in-game/client data or their item list on their site :)

File: db/re/item_db.txt [hunk 1]
```diff
@@ -10759,8 +10774,12 @@
...
+20780,Unity_STR_Manteau,Unity STR Manteau,4,32000,,300,,20,,1,0xFFFFFFFF,63,2,4,,1,1,0,{ .@r = getrefine(); if(BaseLevel<100) { bonus2 bExpAddRace,RC_All,4; bonus bMaxHPrate,2*(.@r/2); } },{},{}
+20781,Unity_AGI_Manteau,Unity AGI Manteau,4,32000,,300,,20,,1,0xFFFFFFFF,63,2,4,,1,1,0,{ .@r = getrefine(); if(BaseLevel<100) { bonus2 bExpAddRace,RC_All,4; bonus bFlee,3*(.@r/2); } },{},{}
+20782,Unity_INT_Muffler,Unity INT Muffler,4,32000,,300,,20,,1,0xFFFFFFFF,63,2,4,,1,1,0,{ .@r = getrefine(); if(BaseLevel<100) { bonus2 bExpAddRace,RC_All,4; bonus bUseSPrate,-(.@r/2); } },{},{}
 20783,Hero_Cape,Hero Cape,4,0,,100,,,,1,0xFFFFFFFF,63,2,4,1,10,1,,{ .@r = getrefine(); bonus2 bSubEle,Ele_Neutral,1+(.@r/3); bonus bFlee,1+(.@r/3); if (.@r > 9) { bonus2 bSubEle,Ele_Neutral,20; bonus bFlee,20; } },{},{}
 20788,Private_Doram_Manteau,Private Doram Manteau,4,20,,300,,20,,1,0x80000000,7,2,4,,100,1,,{ bonus bFlee,5; bonus2 bSubEle,Ele_Neutral,2; bonus bFlee2,getrefine()/3; },{},{}
+20787,Unity_Exquisite_Muffler,Unity Exquisite Muffler,4,32000,,300,,20,,1,0xFFFFFFFF,63,2,4,,1,1,0,{ .@r = getrefine(); if(BaseLevel<100) { bonus2 bExpAddRace,RC_All,4; bonus bCritical,2*(.@r/2); } },{},{}
```
Comment (Lemongrass3110) [inline db/re/item_db.txt]: Wrong order

File: db/re/item_combo_db.txt [hunk 2]
```diff
@@ -469,11 +478,43 @@
...
+20718:22002,{ bonus bMaxHPRate,15; bonus bMaxSPRate,5; }
+20718:22003,{ bonus bMaxHPRate,15; bonus bMaxSPRate,5; }
+20718:22004,{ bonus bMaxHPRate,15; bonus bMaxSPRate,5; }
+20718:22005,{ bonus bMaxHPRate,15; bonus bMaxSPRate,5; }
+20718:22006,{ bonus bMaxHPRate,15; bonus bMaxSPRate,5; }
+20718:22007,{ bonus bMaxHPRate,15; bonus bMaxSPRate,5; }
+20718:22008,{ bonus bMaxHPRate,15; bonus bMaxSPRate,5; }
+20718:22009,{ bonus bMaxHPRate,15; bonus bMaxSPRate,5; }
+20718:22010,{ bonus bMaxHPRate,15; bonus bMaxSPRate,5; }
+22011:20718,{ bonus bMaxHPRate,15; bonus bMaxSPRate,5; }
```
Comment (zackdreaver) [inline db/re/item_combo_db.txt]: Yeah I swap them, for sorting purpose, will do follow up

File: db/re/item_db.txt [hunk 3]
```diff
@@ -2169,7 +2172,7 @@
 2987,Snake_Pendant,Snake Pendant,4,20,,100,,3,,1,0xFFFFFFFF,63,2,136,,0,0,0,{ bonus bAgi,3; bonus bLuk,2; bonus bMdef,3; },{},{}
 2988,Ozs_New_Wing_Ring,Oz's New Wing Ring,4,20,,100,,0,,1,0x00080000,56,1,136,,130,,,{ bonus bVariableCastrate,-25; },{},{}
 2989,Bloody_Floral_Decoration_Bracelet,Bloody Floral Decoration Bracelet,4,20,,100,,0,,1,0x00080000,56,0,136,,130,,,{ bonus bVariableCastrate,-25; },{},{}
-2990,Pendant_of_Harmony,Pendant of Harmony,4,20,,100,,0,,0,0xFFFFFFFF,63,2,136,,130,,,{ bonus bMatkRate,6; bonus bHPrecovRate,50; bonus bSPrecovRate,50; bonus2 bSubEle,Ele_Water,5; bonus2 bSubEle,Ele_Earth,5; bonus2 bSubEle,Ele_Fire,5; bonus2 bSubEle,Ele_Wind,5; bonus2 bSubEle,Ele_Poison,5; bonus2 bSubEle,Ele_Ghost,5; bonus2 bSubEle,Ele_Undead,5; bonus4 bAutoSpellWhenHit,"PR_SANCTUARY",3,100,0; bonus3 bAutoSpellWhenHit,"AB_RENOVATIO",1,100; bonus bStr,-5; },{},{}
+2990,Pendant_of_Harmony,Pendant of Harmony,4,20,,100,,0,,0,0xFFFFFFFF,63,2,136,,130,,,{ bonus bStr,-5; bonus bMatkRate,6; bonus bHPrecovRate,50; bonus bSPrecovRate,50; bonus2 bSubEle,Ele_Holy,5; bonus5 bAutoSpell,"AB_RENOVATIO",1,100,BF_NORMAL,0; bonus4 bAutoSpellWhenHit,"PR_SANCTUARY",3,100,0; },{},{}
```
Comment (zackdreaver) [inline db/re/item_db.txt]: Ok, in next follow up

File: db/re/item_combo_db.txt [hunk 4]
```diff
@@ -490,7 +532,7 @@
 22124:28467:28468,{ bonus bAspdRate,8; bonus bHit,8; bonus2 bExpAddClass,Class_All,2; }
 22125:28469:28470,{ bonus bAspdRate,10; bonus bHit,10; bonus2 bExpAddClass,Class_All,2; }
 22126:28471:28472,{ bonus bAspdRate,10; bonus bHit,10; bonus2 bExpAddClass,Class_All,2; bonus bSpeedAddRate,25; }
-24012:24013:24014:24015:24016:24017,{ bonus bAllStats,9; }
+24012:24013:24014:24015:24016:24017,{ bonus bAllStats,3; }
```
Comment (secretdataz) [inline db/re/item_combo_db.txt]: I don't see a kRO description indicating this combo. Link to the source is appreciated.

File: db/re/item_db.txt [hunk 5]
```diff
@@ -1215,6 +1215,7 @@
 1938,Infinity_Violin,Infinity Violin,5,10,,500,150,,2,1,0x00080000,56,1,2,4,100,1,13,{},{},{}
 1939,Crimson_Violin,Crimson Violin,5,20,,800,80,,2,2,0x00080000,63,1,2,3,70,1,13,{ .@r = getrefine(); bonus bBaseAtk,((BaseLevel/10)*5)+(.@r<=15?pow(.@r,2):225); },{},{}
 1940,Trumpet_Shell,Trumpet Shell,5,0,,600,125,,1,1,0x00080000,56,1,2,4,130,1,13,{ .@r = getrefine(); bonus bDex,.@r; bonus bInt,.@r; .@per = ((.@r > 9) ? 40 : ((.@r > 7) ? 20 : ((.@r > 5) ? 10 : 0))); bonus2 bSkillAtk,"WM_REVERBERATION",.@per; bonus2 bVariableCastrate,"WM_REVERBERATION",-.@per; },{},{}
+1941,Unity_Violin,Unity Violin,5,20,,400,76,,2,1,0x00080000,63,1,2,3,1,1,13,{ .@r = getrefine()*getrefine(); bonus bBaseAtk,.@r*=125/100; },{},{}
```
Comment (secretdataz) [inline db/re/item_db.txt]: `bonus bBaseAtk,pow(getrefine(),2)*125/100;`

File: db/re/item_db.txt [hunk 6]
```diff
@@ -1265,6 +1266,7 @@
 1994,Infinity_Whip,Infinity Whip,5,10,,500,150,,2,1,0x00080000,56,0,2,4,100,1,14,{},{},{}
 1995,Crimson_Whip,Crimson Whip,5,20,,800,80,,2,2,0x00080000,63,0,2,3,70,1,14,{ .@r = getrefine(); bonus bBaseAtk,((BaseLevel/10)*5)+(.@r<=15?pow(.@r,2):225); },{},{}
 1996,Wire_of_Vicious_Mind,Wire of Vicious Mind,5,20,,1300,130:50,,2,1,0x00080000,63,,2,4,160,1,14,{ bonus bAtk,pow(min(getrefine(),15),2); bonus bMatk,pow(min(getrefine(),15),2)/2; },{},{}
+1997,Unity_Whip,Unity Whip,5,20,,76,40,,2,1,0x00080000,63,0,2,3,1,1,14,{ .@r = getrefine()*getrefine(); bonus bBaseAtk,.@r*=125/100; },{},{}
```
Comment (secretdataz) [inline db/re/item_db.txt]: bonus bBaseAtk,pow(getrefine(),2)*125/100;

File: db/re/item_db.txt [hunk 7]
```diff
@@ -1296,6 +1298,7 @@
 2025,Crimson_Two-Handed_Staff,Crimson Two-Handed Staff,5,20,,1000,100,,1,2,0x800200,63,2,34,3,70,1,23,{ .@r = getrefine(); bonus bInt,5; bonus bMatk,150+((BaseLevel/10)*5)+(.@r<=15?pow(.@r,2):225); },{},{}
 2026,Staff_of_Vicious_Mind,Staff of Vicious Mind,5,20,,1000,100:200,,1,1,0x00810204,63,2,34,4,160,1,23,{ bonus bMatk,pow(min(getrefine(),15),2); bonus bInt,6; bonus bUnbreakableWeapon,1; },{},{}
 2027,Sunflower_Kid,Sunflower Kid,5,10,,1500,150:320,,,2,0x00000200,40,2,34,4,170,1,23,{ bonus bInt,5; .@r = getrefine(); bonus bMatkRate,(.@r/2); bonus bDelayrate,(.@r*-1); bonus bUnbreakableWeapon,1; },{},{}
+2030,Unity_Two-Handed_Staff,Unity Two-Handed Staff,5,20,,500,95:135,,1,1,0x800200,63,2,34,3,1,1,23,{ .@r = getrefine()*getrefine(); bonus bMatk,.@r*=125/100; },{},{}
```
Comment (secretdataz) [inline db/re/item_db.txt]: bonus bBaseAtk,pow(getrefine(),2)*125/100;

File: db/re/item_db.txt [hunk 8]
```diff
@@ -7243,10 +7247,8 @@
 13125,Metal_Revolver,Metal Revolver,5,0,,0,30,,7,1,0x41000000,63,2,34,3,1,0,17,{ bonus bBaseAtk,getrefine(); bonus bLongAtkRate,1; if (BaseLevel >= 20 && BaseLevel <= 120) { bonus bBaseAtk,3*(BaseLevel/10); } },{},{}
 13126,Infinity_Pistol,Infinity Pistol,5,10,,500,175,,7,1,0x40000000,63,2,34,4,100,0,17,{},{},{}
 13127,Crimson_Revolver,Crimson Revolver,5,20,,1000,100,,7,2,0x41000000,63,2,34,3,70,1,17,{ .@r = getrefine(); bonus bBaseAtk,((BaseLevel/10)*5)+(.@r<=15?pow(.@r,2):225); },{},{}
-//===================================================================
-// Guns
-//===================================================================
 13128,Revolver_of_Vicious_Mind,Revolver of Vicious Mind,5,20,,1500,150,,7,1,0x41000000,63,2,34,4,160,1,17,{ bonus bAtk,pow(min(getrefine(),15),2); },{},{}
+13129,Unity_Revolver,Unity Revolver,5,20,,500,95,,7,1,0x41000000,63,2,34,3,1,1,17,{ .@r = getrefine()*getrefine(); bonus bBaseAtk,.@r*=125/100; },{},{}
```
Comment (secretdataz) [inline db/re/item_db.txt]: bonus bBaseAtk,pow(getrefine(),2)*125/100;

File: db/re/item_db.txt [hunk 9]
```diff
@@ -7406,6 +7408,7 @@
 13323,Infinity_Shuriken,Infinity Shuriken,5,0,,500,150:40,,1,1,0x20000000,63,2,34,4,100,1,22,{},{},{}
 13327,Crimson_Huuma_Shuriken,Crimson Huuma Shuriken,5,20,,1000,100,,1,2,0x22000000,63,2,34,3,70,1,22,{ .@r = getrefine(); bonus bBaseAtk,((BaseLevel/10)*5)+(.@r<=15?pow(.@r,2):225); bonus bMatk,(.@r<=15?(pow(.@r,2)/2):225); },{},{}
 13328,Huuma_Shuriken_of_Vicious_Mind,Huuma Shuriken of Vicious Mind,5,20,,1500,150:50,,1,1,0x22000000,63,2,34,4,160,1,22,{ bonus bAtk,pow(min(getrefine(),15),2); bonus bMatk,pow(min(getrefine(),15),2)/2; bonus bUnbreakableWeapon,1; },{},{}
+13329,Unity_Huuma_Shuriken,Unity Huuma Shuriken,5,20,,500,95,,1,1,0x22000000,63,2,34,3,1,1,22,{ .@r = getrefine()*getrefine(); bonus bBaseAtk,.@r*=125/100; },{},{}
```
Comment (secretdataz) [inline db/re/item_db.txt]: bonus bBaseAtk,pow(getrefine(),2)*125/100;

File: db/re/item_db.txt [hunk 10]
```diff
@@ -7457,6 +7460,8 @@
 13452,Ru_Gold_Sword,Ru Gold Sword,5,0,,1200,190,,1,2,0x00000080,56,2,2,3,120,1,2,{ bonus bStr,8; bonus bAgi,8; },{},{}
 13454,Crimson_Saber,Crimson Saber,5,20,,850,85,,1,2,0x000654E3,56,2,2,3,70,1,2,{ .@r = getrefine(); bonus bBaseAtk,((BaseLevel/10)*5)+(.@r<=15?pow(.@r,2):225); },{},{}
 13455,Saber_of_Vicious_Mind,Saber of Vicious Mind,5,20,,1350,135,,1,1,0x000654E3,63,2,2,4,160,1,2,{ bonus bAtk,pow(min(getrefine(),15),2); },{},{}
+13456,Unity_Sword,Unity Sword,5,20,,420,80,,1,1,0x000654E3,56,2,2,3,1,1,2,{ .@r = getrefine()*getrefine(); bonus bBaseAtk,.@r*=125/100; },{},{}
```
Comment (secretdataz) [inline db/re/item_db.txt]: bonus bBaseAtk,pow(getrefine(),2)*125/100;

File: db/re/item_db.txt [hunk 11]
```diff
@@ -7457,6 +7460,8 @@
...
+13457,Unity_Dagger,Unity Dagger,5,20,,270,52,,1,1,0x028F5EEF,63,2,2,3,1,1,1,{ .@r = getrefine()*getrefine(); bonus bBaseAtk,.@r*=125/100; },{},{}
```
Comment (secretdataz) [inline db/re/item_db.txt]: bonus bBaseAtk,pow(getrefine(),2)*125/100;

File: db/re/item_db.txt [hunk 12]
```diff
@@ -8690,6 +8699,7 @@
 16040,Crimson_Mace,Crimson Mace,5,20,,800,80,,1,2,0x0004C5B3,63,2,2,3,70,1,8,{ .@r = getrefine(); bonus bBaseAtk,((BaseLevel/10)*5)+(.@r<=15?pow(.@r,2):225); },{},{}
 16041,Mace_of_Vicious_Mind,Mace of Vicious Mind,5,20,,1300,130,,1,1,0x0004C5B3,63,2,2,4,160,1,8,{ bonus bAtk,pow(min(getrefine(),15),2); bonus bUnbreakableWeapon,1; },{},{}
 16043,Meteor_Strike,Meteor Strike,5,0,,20000,1,,1,2,0x00000001,63,2,2,4,110,1,8,{ bonus bBaseAtk,10*getskilllv("BS_WEAPONRESEARCH"); bonus bBaseAtk,30*getskilllv("MO_IRONHAND"); .@s = getskilllv("AM_AXEMASTERY"); bonus bBaseAtk,7*.@s; bonus bHit,5*.@s; bonus bBaseAtk,10*getrefine(); if (getskilllv("MC_PUSHCART") > 9) skill "MC_CARTREVOLUTION",1; if (getskilllv("SM_SWORD") > 0) skill "KN_BOWLINGBASH",1; .@str = readparam(bStr); if (.@str > 119) bonus bUseSPrate,-30; else if (.@str > 107) bonus bUseSPrate,-20; },{},{}
+16048,Unity_Mace,Unity Mace,5,20,,400,76,,1,1,0x0004C5B3,63,2,2,3,1,1,8,{ .@r = getrefine()*getrefine(); bonus bBaseAtk,.@r*=125/100; },{},{}
```
Comment (secretdataz) [inline db/re/item_db.txt]: bonus bBaseAtk,pow(getrefine(),2)*125/100;

File: db/re/item_db.txt [hunk 13]
```diff
@@ -9419,6 +9429,7 @@
 18127,Ru_Gold_Bow,Ru Gold Bow,5,0,,1200,150,,5,2,0x00000800,56,2,34,3,120,1,11,{ bonus bAgi,8; bonus bDex,8; },{},{}
 18128,Infinity_Bow,Infinity Bow,5,20,,500,160,,5,1,0x00020800,56,2,34,4,100,1,11,{},{},{}
 18130,Crimson_Bow,Crimson Bow,5,20,,1200,120,,5,2,0x000A0848,63,2,34,3,70,1,11,{ .@r = getrefine(); bonus bBaseAtk,((BaseLevel/10)*5)+(.@r<=15?pow(.@r,2):225); },{},{}
+18132,Unity_Bow,Unity Bow,5,20,,600,114,,5,2,0x000A0848,63,2,34,3,1,1,11,{ .@r = getrefine()*getrefine(); bonus bBaseAtk,.@r*=125/100; },{},{}
```
Comment (secretdataz) [inline db/re/item_db.txt]: bonus bBaseAtk,pow(getrefine(),2)*125/100;

File: db/re/item_db.txt [hunk 14]
```diff
@@ -9754,6 +9765,7 @@
 18872,Snake_Hat,Snake Hat,4,20,,100,,5,,1,0xFFFFFFFF,63,2,256,,0,1,986,{ bonus bStr,3; bonus bInt,3; bonus bMdef,5; bonus bUnbreakableHelm,1; },{},{}
 18873,Sweet_Valentine_Out,Sweet Valentine Out,4,20,,100,,1,,0,0xFFFFFFFF,63,2,256,,1,1,865,{ bonus bMaxHPrate,7; bonus bMaxSPrate,7; },{},{}
 18874,Cyclops_Glasses,Cyclops Glasses,4,10,,100,,,,1,0xFFFFFFFF,63,2,512,,1,,23,{},{},{}
+18877,Baron_Evil_Eye,Baron's Evil Eye,4,10,,300,,,,0,0xFFFFFFFF,63,2,512,,30,,989,{ bonus bDelayrate,5; bonus bUseSPrate,5; },{},{}
```
Comment (secretdataz) [inline db/re/item_db.txt]: bDelayrate should be -5

File: db/re/item_db.txt [hunk 15]
```diff
@@ -11351,29 +11405,43 @@
 28007,Crimson_Katar,Crimson Katar,5,20,,1300,130,,1,2,0x00001000,63,2,34,3,70,1,16,{ .@r = getrefine(); bonus bBaseAtk,((BaseLevel/10)*5)+(.@r<=15?pow(.@r,2):225); },{},{}
 28008,Katar_of_Vicious_Mind,Katar of Vicious Mind,5,20,,1800,180,,1,1,0x00001000,63,2,34,4,160,1,16,{ bonus bAtk,pow(min(getrefine(),15),2); },{},{}
 28010,Juliette_D._Rachel,Juliette D. Rachel,5,20,,2500,300,,1,2,0x00001000,56,2,34,4,170,1,16,{ .@r = getrefine(); bonus bAtkRate,(.@r/2); bonus bAspdRate,.@r; bonus bUnbreakableWeapon,1; },{},{}
-//
-28100,Tanos_Axe_,Tanos Axe,5,10,,4000,300:80,,,1,0x00000400,63,2,2,4,120,1,6,{ bonus bInt,6; bonus bVit,6; bonus bLuk,-6; bonus bAtkRate,5; bonus2 bHPLossRate,100,10000; },{},{}
-28101,Axe_Tornado,Axe Tornado,5,10,,4000,320,,,1,0x00000400,56,2,2,4,130,,6,{ bonus bAtkEle,4; bonus2 bSkillCooldown,"NC_AXETORNADO",-1000; },{},{}
+28011,Unity_Katar,Unity Katar,5,20,,650,123,,1,1,0x00001000,63,2,34,3,1,1,16,{ .@r = getrefine()*getrefine(); bonus bBaseAtk,.@r*=125/100; },{},{}
```
Comment (secretdataz) [inline db/re/item_db.txt]: bonus bBaseAtk,pow(getrefine(),2)*125/100;

File: db/re/item_db.txt [hunk 16]
```diff
@@ -11351,29 +11405,43 @@
...
-28103,Blue_Twohand_Axe,Blue Twohand Axe,5,10,,3000,330,,1,1,0x00000400,56,2,2,3,1,1,7,{ bonus bStr,5; bonus bVit,5; },{},{}
+28103,Blue_Twohand_Axe,Blue Twohand Axe,5,10,,3000,330,,1,1,0x00000400,56,2,34,3,1,1,7,{ bonus bStr,5; bonus bVit,5; },{},{}
 28104,Ru_Gold_Axe_M,Ru Gold Axe M,5,0,,3000,330,,1,2,0x00000400,56,2,34,3,120,1,7,{ bonus bStr,8; bonus bVit,8; },{},{}
 28105,Infinity_Axe,Infinity Axe,5,10,,500,265,,1,1,0x000444A2,56,2,34,4,100,1,7,{},{},{}
 28106,Crimson_Two-Handed_Axe,Crimson Two-Handed Axe,5,20,,2000,200,,1,2,0x000444A2,63,2,34,3,70,1,7,{ .@r = getrefine(); bonus bBaseAtk,((BaseLevel/10)*5)+(.@r<=15?pow(.@r,2):225); bonus bUnbreakableWeapon,1; },{},{}
-//
 28107,Two_Handed_Axe_of_Vicious_Mind,Two Handed Axe of Vicious Mind,5,20,,2500,250,,1,1,0x000444A2,63,2,34,4,160,1,7,{ bonus bAtk,pow(min(getrefine(),15),2); bonus bUnbreakableWeapon,1; },{},{}
+28110,Unity_Two-Handed_Axe,Unity Two-Handed Axe,5,20,,1000,190,,1,1,0x000444A2,63,2,34,3,1,1,7,{ .@r = getrefine()*getrefine(); bonus bBaseAtk,.@r*=125/100; },{},{}
```
Comment (secretdataz) [inline db/re/item_db.txt]: bonus bBaseAtk,pow(getrefine(),2)*125/100;

File: db/re/item_db.txt [hunk 17]
```diff
@@ -11351,29 +11405,43 @@
...
-//
+//===================================================================
+// New Accessories
+//===================================================================
+28302,Vesper_Core01_,Vesper Core 01,4,20,,500,,1,,1,0x00CFFF80,18,2,136,,1,0,0,{ bonus bMdef,3; bonus bInt,2; bonus bMaxSPrate,5; },{},{}
+28303,Vesper_Core02_,Vesper Core 02,4,20,,500,,1,,1,0x00CFFF80,18,2,136,,1,0,0,{ bonus bMdef,3; bonus bStr,3; bonus bBaseAtk,10; },{},{}
+28304,Vesper_Core03_,Vesper Core 03,4,20,,500,,1,,1,0x00CFFF80,18,2,136,,1,0,0,{ bonus bMdef,3; bonus bAgi,3; bonus bFlee,5; },{},{}
+28305,Vesper_Core04_,Vesper Core 04,4,20,,500,,1,,1,0x00CFFF80,18,2,136,,1,0,0,{ bonus bMdef,3; bonus bDex,3; bonus bHit,10; },{},{}
+28306,Bless_Of_Moon,Luna's Blessing,4,20,,100,,5,,0,0xFFFFFFFF,63,2,136,,1,0,,{ bonus bAllStats,1; bonus bMdef,2; bonus3 bAutoSpell,"CG_TAROTCARD",5,100; },{},{}
```
Comment (secretdataz) [inline db/re/item_db.txt]: Combo script for 28302-28306 missing?

File: db/re/item_db.txt [hunk 18]
```diff
@@ -11387,6 +11455,8 @@
 28381,Cute_Grass_Necklace,Cute Grass Necklace,4,20,,10,,0,,1,0x80000000,7,2,136,,100,0,,{ bonus bHealPower,5; bonus2 bSkillCooldown,"SU_TUNAPARTY",-7000; },{},{}
 28382,Charm_Grass_Necklace,Charm Grass Necklace,4,20,,10,,0,,1,0x80000000,7,2,136,,100,0,,{ bonus bMdef,5; bonus2 bVariableCastrate,"SU_CN_METEOR",-1000; },{},{}
 28386,Fallen_Monk_Rosary,Fallen Monk Rosary,4,0,,100,,,,1,0xFFFFFFFF,63,2,136,1,20,0,,{ bonus bAspdRate,5; bonus bCritical,5; bonus2 bSkillUseSP,"CH_SOULCOLLECT",-5; if (BaseLevel > 14) bonus2 bSkillAtk,"SR_DRAGONCOMBO",BaseLevel/15; if (BaseLevel > 19) bonus2 bSkillAtk,"SR_KNUCKLEARROW",BaseLevel/20; if (BaseLevel > 29) bonus2 bSkillAtk,"SR_SKYNETBLOW",BaseLevel/30; },{},{}
+28391,S_Thief_Earring,Thief Shadow Earring,12,0,,0,,,,0,0xFFFFFFFF,63,2,1048576,,1,1,,{ autobonus3 "{ bonus bBaseAtk,30l bonus bFlee,30; }",200,5000,"TF_THROWSTONE","{ specialeffect2 EF_ENHANCE; }"; },{},{}
```
Comment (secretdataz) [inline db/re/item_db.txt]: Doesn't match "돌던지기 사용시, 20%의 확률로 5초간 ATK와 FLEE 30증가. 제련도 1당 ATK와 FLEE 10 추가 증가".
Please show your source for your implemented script.

File: db/re/item_db.txt [hunk 19]
```diff
@@ -11387,6 +11455,8 @@
...
+28392,S_Archer_Earring,Archer Shadow Earring,12,0,,0,,,,0,0xFFFFFFFF,63,2,1048576,,1,1,,{ .@r = getrefine(); autobonus3 "{ bonus bBaseAtk,30+.@r; bonus bHit,30+.@r; }",200,5000,"AC_DOUBLE","{ specialeffect2 EF_ENHANCE; }"; },{},{}
```
Comment (secretdataz) [inline db/re/item_db.txt]: Doesn't match kRO description.

File: db/re/item_db.txt [hunk 20]
```diff
@@ -11432,24 +11502,46 @@
...
+//===================================================================
+// More books
+//===================================================================
 28600,Blue_Book,Blue Book,5,10,,500,160,,1,1,0x00000100,56,2,2,3,100,1,15,{ bonus bAgi,5; bonus bDex,5; },{},{}
 28601,Ru_Gold_Book,Ru Gold Book,5,0,,500,160,,1,2,0x00000008,63,2,2,3,120,1,15,{ bonus bVit,8; bonus bInt,8; },{},{}
 28602,Demon_Hunting_Bible,Demon Hunting Bible,5,0,,500,30:170,,1,2,0x00000008,63,2,2,3,110,1,15,{ bonus bInt,2; bonus bDex,2; .@b = readparam(bInt); bonus2 bSkillAtk,"PR_MAGNUS",30+min(.@b,120); },{},{}
 28604,Crimson_Bible,Crimson Bible,5,20,,450,45,,1,2,0x00410100,63,2,2,3,70,1,15,{ .@r = getrefine(); bonus bBaseAtk,((BaseLevel/10)*5)+(.@r<=15?pow(.@r,2):225); bonus bMatk,(.@r<=15?(pow(.@r,2)/2):225); },{},{}
 28605,Book_of_Vicious_Mind,Book of Vicious Mind,5,20,,950,95,,1,1,0x00010100,63,2,2,4,160,1,15,{ bonus bAtk,pow(min(getrefine(),15),2); bonus bMatk,pow(min(getrefine(),15),2); bonus bUnbreakableWeapon,1; },{},{}
+28606,Unity_Bible,Unity Bible,5,20,,220,42,,1,1,0x00410100,63,2,2,3,1,1,15,{ .@r = getrefine()*getrefine(); bonus bBaseAtk,.@r*=125/100; },{},{}
```
Comment (secretdataz) [inline db/re/item_db.txt]: bonus bBaseAtk,pow(getrefine(),2)*125/100;

File: db/re/item_db.txt [hunk 21]
```diff
@@ -11432,24 +11502,46 @@
...
+28902,Mad_Bunny_,Mad Bunny,4,20,,100,,0,,1,0xFFFFFFFF,63,2,32,,1,1,1,{ bonus2 bAddRace,RC_All,5; bonus2 bMagicAddRace,RC_All,5; bonus bShortWeaponDamageReturn,10; autobonus2 "{ bonus bMagicDamageReturn,60; }",10,2000,BF_MAGIC,"{ specialeffect2 EF_WIND; }"; .@r = getrefine(); if(.@r>=7) { bonus bBaseAtk,5; bonus bMatk,5; } if(.@r>=9) { bonus bBaseAtk,15; bonus bMatk,15; } if(.@r>=12) { bonus bCritical,10; bonus bNoCastCancel,0; } },{},{}
 28903,Scutum,Scutum,4,0,,500,,1,,1,0xFFFFFFFF,63,2,32,1,1,1,,{ .@r = getrefine(); bonus bFlee,5+(.@r*3); bonus bFlee2,1+(.@r*2); if (.@r > 10) { bonus bMaxHPrate,10; bonus bMaxSPrate,10; } },{},{}
+28904,Unity_Guard,Unity Guard,4,20,,300,,60,,1,0xFFFFFFFF,63,2,32,,100,1,1,{ .@r = getrefine(); if(BaseLevel<100) { bonus bDef,12*.@r; } },{},{}
```
Comment (secretdataz) [inline db/re/item_db.txt]: `.@r` assignment isn't needed if there's only a single statement referring to refine value. Also applies to 28905.

File: db/re/item_db.txt (hunk 18 above)
Comment (zackdreaver) [inline db/re/item_db.txt]: I forgot the refine bonuses

Comment (zackdreaver) [inline db/re/item_db.txt]: And for the special effect, it's a custom one i made, no clue what the effect it uses.

File: db/re/item_combo_db.txt (hunk 4 above)
Comment (zackdreaver) [inline db/re/item_combo_db.txt]: It's in the commit message here
https://github.com/rathena/rathena/pull/1848/commits/a4251f61d1e41d9bcc2927160e00b8b0a97b77fd

**Pull request merged**